### PR TITLE
Silence ruby warnings

### PR DESCRIPTION
**Pull request by dougal**
Before:

    $ rake 2>&1 | grep warning | grep propono | wc -l
    104

After:

    $ rake 2>&1 | grep warning | grep propono | wc -l
    0

Includes extra tests added to `configuration_test.rb` to make sure that the refactored configuration class is as expected.

Note: The rewritten Configuration class makes use of keyword arguments internally, and thus is only compatible with Ruby 2+.

Unless Amazon will accept a PR to remove the warnings in their `aws-sdk` and `seahorse` gems, then perhaps it might serve well to turn off warnings for the `rake test` task. Test output is still messy despite removing propono's own warnings.

### CODE REVIEW

**Comment by dougal** [comment]
Side benefit: Code Climate is now [much happier](https://codeclimate.com/github/iHiD/propono/pull/79).

File: test/services/publisher_test.rb [hunk 1]
```diff
@@ -115,12 +115,6 @@ def test_publish_should_raise_exception_if_topic_is_nil
       end
     end
 
-    def test_publish_should_raise_exception_if_topic_is_nil
```
Comment (iHiD) [inline test/services/publisher_test.rb]: @dougal Why did you remove this test?

Comment (dougal) [reply test/services/publisher_test.rb]: It's a duplicate.

https://github.com/iHiD/propono/blob/master/test/services/publisher_test.rb#L112-L122

Comment (iHiD) [reply test/services/publisher_test.rb]: Wonderful. Thanks.
